sway-lib-core/src/raw_ptr.sw

```diff
@@ -49,4 +49,43 @@ impl raw_ptr {
             };
         }
     }
+
+    /// Writes the given byte to the address.
+    pub fn write_byte(self, val: u8) {
+        let val_ptr = asm(r1: val) { r1: raw_ptr };
+        asm(ptr: self, val: val_ptr) {
+            sb ptr val i0;
+        };
+    }
```

Comment:
This worries me a bit.  Has this been tested?

Line 55 works for casting a reference type to an address, because `val` will actually be an implicit pointer.  But when `val` is a copy type, like `u8`, line 55 is just converting the value into a pointer (e.g., if `val` is 10 then you will have an address of 10, not the address of `val`).  At least that's how I think it's working, I'd need to double check.

Reply:
This seems to work according to the in-language tests in std::bytes, but ideally I could use `_addr_of` instead of this cast to a `raw_ptr`.